l10n_pe_toponym/models/l10n_pe_toponym.py

```diff
@@ -0,0 +1,60 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+from odoo.addons.base.models.res_country import location_name_search
+
+
+class L10npeCountryProvince(models.Model):
+    _name = 'l10n_pe.res.country.province'
+    _description = 'Lista de Provincias'
+
+    state_id = fields.Many2one(
+        'res.country.state',
+        string='Departamento',
+        required=True
+    )
+    country_id = fields.Many2one(
+        'res.country',
+        related='state_id.country_id',
+        string='País'
+    )
+    name = fields.Char(string='Nombre', required=True)
+    code = fields.Char(string='Código', required=True)
+
+    name_search = location_name_search
+
+    _sql_constraints = [
+        ('name_code_uniq', 'unique(state_id, code)',
+         'Código de la Provincia debe ser único por Departamento!')
+    ]
+
+
+class L10npeCountryDistrict(models.Model):
+    _name = 'l10n_pe.res.country.district'
+    _description = 'Lista de Distritos'
+
+    province_id = fields.Many2one(
+        'l10n_pe.res.country.province',
+        string='Provincia',
+        required=True
+    )
+    state_id = fields.Many2one(
+        'res.country.state',
+        related='province_id.state_id',
+        string='Departamento'
+    )
+    country_id = fields.Many2one(
```

Comment:
Si el modelo es una localización para Perú se entiede que todos las distritos tendrían como país a Perú, quizas no sea necesario colocarle este atributo country_id.

Reply:
Buena observación, trabajando

Reply:
Aunque realmente sea así, diría de mantenerlo para que podáis tener dominios autómaticos asociados con este campo y no se pueda llegar a la incongruencia de establecer un departamento en un contacto cuando el país no es Perú.

Reply:
OK

Reply:
Actualizado con el cambio discutido.